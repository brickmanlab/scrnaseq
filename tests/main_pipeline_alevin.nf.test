nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"

    test("test-dataset_simpleaf_aligner") {

        when {
            // the rest is taken from shared config
            params {
                aligner    = 'simpleaf'
                outdir     = "${outputDir}/results_simpleaf"
            }
        }

        then {

            assertAll(

                //
                // General assertions
                //

                // Did it finish successfully?
                {assert workflow.success},

                // How many tasks were executed?
                {assert workflow.trace.tasks().size() == 15},

                // How many results were produced?
                {assert path("${outputDir}/results_simpleaf").list().size() == 4},
                {assert path("${outputDir}/results_simpleaf/simpleaf").list().size() == 3},
                {assert path("${outputDir}/results_simpleaf/simpleaf/mtx_conversions").list().size() == 5},
                {assert path("${outputDir}/results_simpleaf/fastqc").list().size() == 12},
                {assert path("${outputDir}/results_simpleaf/multiqc").list().size() == 3},

                //
                // Check if files were produced
                //
                {assert new File( "${outputDir}/results_simpleaf/simpleaf/mtx_conversions/Sample_X/Sample_X_raw_matrix.h5ad" ).exists()},
                {assert new File( "${outputDir}/results_simpleaf/simpleaf/Sample_X/simpleaf_quant/af_quant/alevin/quants_mat_cols.txt" ).exists()},
                {assert new File( "${outputDir}/results_simpleaf/simpleaf/Sample_X/simpleaf_quant/af_quant/alevin/quants_mat_rows.txt" ).exists()},
                {assert new File( "${outputDir}/results_simpleaf/simpleaf/Sample_X/simpleaf_quant/af_quant/alevin/quants_mat.mtx" ).exists()},
                {assert new File( "${outputDir}/results_simpleaf/simpleaf/Sample_X/simpleaf_quant/af_quant/alevin/quants.h5ad" ).exists()},
                {assert new File( "${outputDir}/results_simpleaf/simpleaf/mtx_conversions/Sample_Y/Sample_Y_raw_matrix.h5ad" ).exists()},
                {assert new File( "${outputDir}/results_simpleaf/simpleaf/Sample_Y/simpleaf_quant/af_quant/alevin/quants_mat_cols.txt" ).exists()},
                {assert new File( "${outputDir}/results_simpleaf/simpleaf/Sample_Y/simpleaf_quant/af_quant/alevin/quants_mat_rows.txt" ).exists()},
                {assert new File( "${outputDir}/results_simpleaf/simpleaf/Sample_Y/simpleaf_quant/af_quant/alevin/quants_mat.mtx" ).exists()},
                {assert new File( "${outputDir}/results_simpleaf/simpleaf/Sample_Y/simpleaf_quant/af_quant/alevin/quants.h5ad" ).exists()},

                //
                // Check if files are the same
                //
                {assert snapshot(
                    path( "${outputDir}/results_simpleaf/simpleaf/mtx_conversions/Sample_X/Sample_X_raw_matrix.seurat.rds" ),
                    path( "${outputDir}/results_simpleaf/simpleaf/mtx_conversions/Sample_X/Sample_X_raw_matrix.sce.rds" ),
                    path( "${outputDir}/results_simpleaf/simpleaf/mtx_conversions/Sample_Y/Sample_Y_raw_matrix.seurat.rds" ),
                    path( "${outputDir}/results_simpleaf/simpleaf/mtx_conversions/Sample_Y/Sample_Y_raw_matrix.sce.rds" )
                ).match()}

            ) // end of assertAll()

        }
    }

}
