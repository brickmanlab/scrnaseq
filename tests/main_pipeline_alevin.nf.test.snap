{
    "test-dataset_alevin_aligner": {
        "content": [
            "quants_mat_cols.txt:md5,e9868982c17a330392e38c2a5933cf97",
            "quants_mat.mtx:md5,b8aa7b3c488fd8923de50a3621d4991f",
            "quants_mat_rows.txt:md5,6227df5a13127b71c71fb18cd8574857",
            "quants_mat_cols.txt:md5,e9868982c17a330392e38c2a5933cf97",
            "quants_mat.mtx:md5,54cd12666016adce94c025b2e07f4b02",
            "quants_mat_rows.txt:md5,6b458a7777260ba90eccbe7919df934b",
            "Sample_X_raw_matrix.seurat.rds:md5,708ec66ee15c31c1a09cbaee035a6508",
            "Sample_X_raw_matrix.sce.rds:md5,3bed89cd187a3f5385636fd5196ef42d",
            "Sample_Y_raw_matrix.seurat.rds:md5,3f031ff7c50ee2a2e13ea86319892ee5",
            "Sample_Y_raw_matrix.sce.rds:md5,c83e8e04b8fd4bb4d03313a1348686dc"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.2"
        },
        "timestamp": "2024-11-29T10:17:18.33882659"
    }
}